test/session_pool/session_pool.test.js

```diff
@@ -258,6 +258,26 @@ describe('SessionPool - testing session pool', () => {
         newSessionPool.teardown();
     });
 
+    test('should persist state on teardown', async () => {
+        const persistStateKey = 'my-session-pool';
+        const persistStateKeyValueStoreId = 'my-key-value-store-for-sessions';
+
+        const newSessionPool = await Apify.openSessionPool({
+            maxPoolSize: 1,
+            persistStateKeyValueStoreId,
+            persistStateKey,
+        });
+
+        await newSessionPool.teardown();
+
+        const kvStore = await Apify.openKeyValueStore(newSessionPool.persistStateKeyValueStoreId);
+        const state = await kvStore.getValue(newSessionPool.persistStateKey);
+
+        expect(newSessionPool.persistStateKeyValueStoreId).toBeDefined();
+        expect(newSessionPool.persistStateKey).toBeDefined();
+        expect(state).toBeDefined();
```

Comment:
This won't work. `getValue()` returns `null` when there's nothing saved and `null` will pass the check, because it's not `undefined`.

We should check that there are at least some data inside, not only that it returns "something".

Reply:
The test is fixed - added checking properties of the state object.